addons/snailmail/static/src/messaging/component/message/message_tests.js

```diff
@@ -0,0 +1,578 @@
+        document.body,
+        '.o_NotificationPopover',
+        "notification popover should be open"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_icon',
+        "popover should have one icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_icon .fa-check',
+        "popover should have the sent icon"
+    );
+    assert.equal(
+        document.querySelector('.o_NotificationPopover_icon').textContent.trim(),
+        "Sent",
+        "popover should have the sent text"
+    );
+});
+
+QUnit.test('Canceled', async function (assert) {
+    assert.expect(7);
+
+    await this.start();
+    const message = this.env.entities.Message.create({
+        id: 10,
+        message_type: 'snailmail',
+        notifications: [['insert', [{
+            id: 11,
+            notification_status: 'canceled',
+            notification_type: 'snail',
+        }]]],
+    });
+    await this.createMessageComponent(message);
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a message component"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon',
+        "should display the notification icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon .fa-paper-plane',
+        "icon should represent snailmail"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_Message_notificationIcon').click();
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover',
+        "notification popover should be open"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_icon',
+        "popover should have one icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_icon .fa-trash-o',
+        "popover should have the canceled icon"
+    );
+    assert.equal(
+        document.querySelector('.o_NotificationPopover_icon').textContent.trim(),
+        "Canceled",
+        "popover should have the canceled text"
+    );
+});
+
+QUnit.test('Pending', async function (assert) {
+    assert.expect(7);
+
+    await this.start();
+    const message = this.env.entities.Message.create({
+        id: 10,
+        message_type: 'snailmail',
+        notifications: [['insert', [{
+            id: 11,
+            notification_status: 'ready',
+            notification_type: 'snail',
+        }]]],
+    });
+    await this.createMessageComponent(message);
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a message component"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon',
+        "should display the notification icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon .fa-paper-plane',
```

Comment:
`<i>` should contain `notificationIcon` BEM (omitting similar comments from now, but should apply in whole diff)